crypto/asn1/tasn_new.c

```diff
@@ -199,7 +199,8 @@ static int asn1_item_ex_combine_new(ASN1_VALUE **pval, const ASN1_ITEM *it,
     return 1;
 
  memerr2:
-    ASN1_item_ex_free(pval, it);
+    if (!combine)
+        ASN1_item_ex_free(pval, it);
```

Comment:
Doesn't this switch from mistakenly freeing `pval` itself to leaking the individual fields of `pval`? I'd expect you actually want to call `asn1_item_combine_free(pval, it, combine)`, though the ASN.1 code is incomprehensible to me, so I may just be missing something.

Reply:
Damnit.  You're absolutely right.  And yes, that seems to be the proper call.  Some will react that this function gets opened up to the world, though, but I see no way around that...

Reply:
Keep it in asn1_locl.h I guess? Though yeah I see no way around stomping on the symbol name.

Reply:
#4578 amd #4579